examples/ping-example/Cargo.toml

```diff
@@ -11,3 +11,6 @@ async-trait = "0.1"
 futures = "0.3.27"
 libp2p = { path = "../../libp2p", features = ["async-std", "dns", "macros", "mplex", "noise", "ping", "tcp", "websocket", "yamux"] }
 multiaddr = { version = "0.17.0" }
+
+[[example]]
+name = "ping"
```

Comment:
Why do we need another example here? The entire crate is an example.

We don't use the "cargo" examples (via `--example`) because the facade crate is heavily feature flagged and it is difficult to document the dependencies needed etc.

Reply:
The use of --example ping is suggested in
`https://github.com/libp2p/rust-libp2p/blob/master/libp2p/src/tutorials/ping.rs`
as well as
`https://docs.rs/libp2p/0.51.1/libp2p/tutorials/ping/index.html`
The only two tutorials present for Rust libp2p have a broken example flag.

My changes merely get this example flag working again.

The conflicting info within tutorials/documentation makes it hard for newcomers to understand the structure of the crate and broken examples do not assist.

Reply:
This is just a suggestion as a newcomer to the project. The discrepancy between the Facade example structure vs native cargo --example structure is inherent to confusion. 
Ping.rs is a great tutorial, though it seems it has some outdated portions.
The cargo native --example approach is what was referenced in the tutorial ping.rs, which is why I chose to fix the link from the root of the repo.
Also, when one runs `cargo run --example` from the root of the project, not all examples are in the list of available examples
.
I would suggest that we fully embrace one method or the other, and remove references of the other to prevent confusion for newcomers.

We can fix the --example for each so they all work from the root of the project. Or if it's preferred not to use --example, we may want to add documentation for which path contains an intended example with usage details for each.

A working example should be very obvious and usable immediately after cloning without having to search around the repo.
This will allow others to learn libp2p faster, which will lead to faster adoption.

I like the facade method, but copying the example outside of libp2p repo directory breaks the dependencies since they're referenced via path traversal .. newcomers may not realize this and may waste time / give up.
`

Reply:
Thank you! I totally agree with you that we need to be consistent. We only recently restructured these examples (https://github.com/libp2p/rust-libp2p/pull/3509) and unfortunately missed a few things.

The relative path dependency is unfortunate but in my eyes the best we can do. At least each example is better contained now, can (almost) be copy pasted and compiled individually. Plus, they are all in one place and not scattered across multiple crates.

I would gladly merge PRs that fix these docs :)

Reply:
Happy to help! I apologize I did not see that issue. I wouldn't mind continuing to flesh out documentation as it has been helping me fully understand the crate.
I did also submit a PR on https://github.com/libp2p/docs for the Rust tutorial, although I just removed the line referencing the the --example flag usage.